packages/flutter_tools/lib/src/base/error_handling_io.dart

```diff
@@ -584,15 +648,35 @@ class ErrorHandlingProcessManager extends ProcessManager {
     Encoding stdoutEncoding = io.systemEncoding,
     Encoding stderrEncoding = io.systemEncoding,
   }) {
-    return _runSync(() => _delegate.runSync(
-      command,
-      workingDirectory: workingDirectory,
-      environment: environment,
-      includeParentEnvironment: includeParentEnvironment,
-      runInShell: runInShell,
-      stdoutEncoding: stdoutEncoding,
-      stderrEncoding: stderrEncoding,
-    ), platform: _platform);
+    return _runSync(() {
+      return _processRunSync(
+        _getCommandPath(command, workingDirectory),
+        _getArguments(command),
+        workingDirectory: workingDirectory,
+        environment: environment,
+        includeParentEnvironment: includeParentEnvironment,
+        runInShell: runInShell,
+        stdoutEncoding: stdoutEncoding,
+        stderrEncoding: stderrEncoding,
+      );
+    }, platform: _platform);
+  }
+
+  String _getCommandPath(List<dynamic> command, String workingDirectory, { bool strict = false }) {
+    final String executable = command.first.toString();
+    final Map<String, String> workingDirectoryCache = _resolvedExecutables[workingDirectory] ??= <String, String>{};
+    return workingDirectoryCache[executable] ??= resolveExecutablePath(
+      executable,
+      workingDirectory,
+      fileSystem: _fileSystem,
+      platform: _platform,
+      logger: _logger,
+      strict: strict,
+    );
+  }
+
+  List<String> _getArguments(List<dynamic> command) {
+    return <String>[for (dynamic arg in command.skip(1)) arg.toString()];
```

Comment:
Not entirely convinced that the new List literal syntax improves things here, but it's up to you.
```dart
String toString(dynamic arg) => arg.toString();
return command.skip(1).map(toString).toList();
```

Reply:
It's to write Iterables after seeing the generated code.

Reply:
Sorry, I don't follow.

Reply:
oops, i typo'd. I meant to type out: "It is hard to write iterables after seeing the generated code"